spec/controllers/api/v1/book_controller_spec.rb

```diff
@@ -2,5 +2,47 @@
 
 require 'rails_helper'
 
-RSpec.describe Api::V1::BooksController, type: :controller do
+describe Api::V1::BooksController, type: :controller do
+  include_context 'Authenticated User'
+
+  describe 'GET #index' do
+    context 'When fetching all books' do
+      let!(:books) { create_list(:book, 3) }
+
+      before do
+        get :index, params: { id: user.id }
+      end
+
+      it 'responses with the books json' do
+        expected = ActiveModel::Serializer::CollectionSerializer.new(
+          books, each_serializer: BookSerializer
+        ).to_json
+        expect(JSON(response.body)['page']).to include(JSON(expected).first)
+      end
+
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+  end
+
+  describe 'GET #show' do
+    context 'When fetching a book' do
+      let!(:book) { create(:book) }
+
+      before do
+        get :show, params: { id: book.id }
+      end
+
+      it 'responses with the user rent json' do
+        expect(JSON(response.body.to_json)).to eq BookSerializer.new(
+          book, root: false
+        ).to_json
```

Comment:
en teoría está bien pero creo que lo podes hacer más sencillo usando el operador `=~` (pegale una leida a lo que hace [aquí](https://relishapp.com/rspec/rspec-expectations/v/2-8/docs/built-in-matchers/operator-matchers)
Creo que podría quedar algo así        ` expect(response.body) =~ Api::V1::BookSerializer.new(book).to_json`